### ISSUE TITLE
Cannot get more than 0 rewards

### ISSUE THREAD

**Issue by CrazySssst** [facebookresearch/impact-driven-exploration#11]
Hello,
 thanks for your codes
I can't get more than 0 rewards  in all environments
I test Count agent and RIDE agent.  they both get any rewards. I guess RND/ICM agent are also bad.
I must be mistaken somewhere, please help me.

cuda:10.1
nvidia driver 418.56
pytorch 1.5.1
torch 1.5.1
torchvision 0.10.1
cudatoolkit: 10.1

For code:
i follow https://github.com/facebookresearch/impact-driven-exploration/issues/6#issuecomment-757122754 , modify dtype=torch.uint8 to dtype=torch.bool 

For hyperparameters:
I modify epsilon to 0.01 and keep others unchanged
I follow values in paper to modify the entropy_coef and intrinsic_reward_coef

Thanks!

**Comment by rraileanu** [comment]
It might be related to the --total_num_frames flag. You need to use 3e7 for MultiRoom-N7-S4, MultiRoom-NoisyTV-N7-S4, MultiRoom-N7-S8, MultiRoom-N10-S4, KeyCorridor-S3-R3, and 1e8 for the other environments. 

Can you provide more information on which tasks you are using or the exact commands used to run the experiments?

**Comment by rraileanu** [comment]
I've added precise instructions for training RIDE on all MiniGrid environments in the README. Let me know if you are still not able to reproduce those results or any other ones from the paper.

**Comment by CrazySssst** [comment]
Thank you for your reply!
I will report the results after I complete experiments according to the hyper-parameters you gave,

**Comment by CrazySssst** [comment]
Hi,  I downgrade my pytorch to  1.1.0 by following the code:
`conda install pytorch==1.1.0 torchvision==0.3.0 cudatoolkit=9.0 -c pytorch
`
   I still can't get any meaningful results
Here is my hyper-parameters
```
`    "args": {
        "alpha": 0.99,
        "baseline_cost": 0.5,
        "batch_size": 32,
        "checkpoint_num_frames": 10000000,
        "disable_checkpoint": false,
        "disable_cuda": false,
        "discounting": 0.99,
        "entropy_cost": 0.0005,
        "env": "MiniGrid-MultiRoom-N7-S4-v0",
        "env_seed": 1,
        "epsilon": 0.01,
        "fix_seed": false,
        "forward_loss_coef": 10.0,
        "intrinsic_reward_coef": 0.1,
        "inverse_loss_coef": 0.1,
        "learning_rate": 0.0001,
        "max_grad_norm": 40.0,
        "model": "ride",
        "momentum": 0,
        "no_reward": false,
        "num_actors": 40,
        "num_buffers": 80,
        "num_input_frames": 1,
        "num_threads": 4,
        "queue_timeout": 1,
        "rnd_loss_coef": 0.1,
        "run_id": 0,
        "save_interval": 10,
        "savedir": "/data/ride/s3r3_0.1int_ent0.0005_3e7_run0",
        "seed": 0,
        "total_frames": 30000000,
        "unroll_length": 100,
        "use_fullobs_intrinsic": false,
        "use_fullobs_policy": false,
        "xpid": "torchbeast-20210413-195418"`
```

This is the log information of the last few rounds of updates 
```
`[INFO:61189 ride:383 2021-04-13 23:01:48,834] After 29952000 frames: loss 23.294344 @ 2557.4 fps. Stats:
{'baseline_loss': 0.0014378174673765898,
 'entropy_loss': -0.09338700771331787,
 'forward_dynamics_loss': 4.379179954528809,
 'inverse_dynamics_loss': 18.625837326049805,
 'mean_control_rewards': 0.002788395155221224,
 'mean_count_rewards': 0.5369794964790344,
 'mean_episode_return': 0.0,
 'mean_intrinsic_rewards': 0.0001950176083482802,
 'mean_rewards': 0.0,
 'mean_total_rewards': 0.0001950176083482802,
 'pg_loss': 0.38127627968788147,
 'total_loss': 23.294343948364258}
[INFO:61189 ride:383 2021-04-13 23:01:53,840] After 29964800 frames: loss 23.598366 @ 2557.8 fps. Stats:
{'baseline_loss': 0.0013506440445780754,
 'entropy_loss': -0.09334824234247208,
 'forward_dynamics_loss': 4.8648858070373535,
 'inverse_dynamics_loss': 18.5413875579834,
 'mean_control_rewards': 0.0032070076558738947,
 'mean_count_rewards': 0.5599256157875061,
 'mean_episode_return': 0.0,
 'mean_intrinsic_rewards': 0.00022072953288443387,
 'mean_rewards': 0.0,
 'mean_total_rewards': 0.00022072953288443387,
 'pg_loss': 0.284089595079422,
 'total_loss': 23.598365783691406}
[INFO:61189 ride:383 2021-04-13 23:01:58,847] After 29977600 frames: loss 22.791515 @ 2557.3 fps. Stats:
{'baseline_loss': 0.0009031041408888996,
 'entropy_loss': -0.09342104941606522,
 'forward_dynamics_loss': 4.074231147766113,
 'inverse_dynamics_loss': 18.726694107055664,
 'mean_control_rewards': 0.0023732478730380535,
 'mean_count_rewards': 0.5275853872299194,
 'mean_episode_return': 0.0,
 'mean_intrinsic_rewards': 0.0001589618914294988,
 'mean_rewards': 0.0,
 'mean_total_rewards': 0.0001589618914294988,
 'pg_loss': 0.08310824632644653,
 'total_loss': 22.791515350341797}
[INFO:61189 ride:383 2021-04-13 23:02:03,850] After 29993600 frames: loss 29.729870 @ 3199.2 fps. Stats:
{'baseline_loss': 0.0009369825129397213,
 'entropy_loss': -0.0934230238199234,
 'forward_dynamics_loss': 11.233283042907715,
 'inverse_dynamics_loss': 18.683874130249023,
 'mean_control_rewards': 0.002366043860092759,
 'mean_count_rewards': 0.5290678143501282,
 'mean_episode_return': 0.0,
 'mean_intrinsic_rewards': 0.0001592515764059499,
 'mean_rewards': 0.0,
 'mean_total_rewards': 0.0001592515764059499,
 'pg_loss': -0.09480264037847519,
 'total_loss': 29.729869842529297}
[INFO:61189 ride:383 2021-04-13 23:02:08,856] After 30003200 frames: loss 23.488457 @ 1918.1 fps. Stats:
{'baseline_loss': 0.0014436335768550634,
 'entropy_loss': -0.09341328591108322,
 'forward_dynamics_loss': 4.683372497558594,
 'inverse_dynamics_loss': 18.619861602783203,
 'mean_control_rewards': 0.0029219742864370346,
 'mean_count_rewards': 0.5460314750671387,
 'mean_episode_return': 0.0,
 'mean_intrinsic_rewards': 0.00019831600366160274,
 'mean_rewards': 0.0,
 'mean_total_rewards': 0.00019831600366160274,
 'pg_loss': 0.27719229459762573,
 'total_loss': 23.48845672607422}`
```

**Comment by CrazySssst** [comment]
I ignore the following code in requestment.txt

`-e git+https://github.com/maximecb/gym-minigrid.git@b84d99a2fbb481977172a85661eee812f022f130#egg=gym_minigrid `
then follow your code 
```
cd impact-driven-exploration
pip install -r requirements.txt

# install MiniGrid
cd gym-minigrid
python setup.py install
```
Could this be the reason?

**Comment by CrazySssst** [comment]
I run RIDE in MultiRoom-N4-S5 can get about 0.7 rewards. 
```
python main.py --env MiniGrid-MultiRoom-N4-S5-v0 --intrinsic_reward_coef 0.5 --entropy_cost 0.001 --run_id=0  --model=ride --total_frames=100000000
```

```
[INFO:3069 ride:383 2021-04-14 14:15:32,577] After 39056000 frames: loss 15.538141 @ 3199.3 fps. Stats:
{'baseline_loss': 0.15743252635002136,
 'entropy_loss': -0.012983767315745354,
 'forward_dynamics_loss': 5.20261812210083,
 'inverse_dynamics_loss': 9.912681579589844,
 'mean_control_rewards': 0.006289934739470482,
 'mean_count_rewards': 0.9735113978385925,
 'mean_episode_return': 0.6990131735801697,
 'mean_intrinsic_rewards': 0.003069306490942836,
 'mean_rewards': 0.02490234375,
 'mean_total_rewards': 0.027971647679805756,
 'pg_loss': 0.2783932685852051,
 'total_loss': 15.538141250610352}
[INFO:3069 ride:383 2021-04-14 14:15:37,584] After 39068800 frames: loss 15.413479 @ 2557.4 fps. Stats:
{'baseline_loss': 0.19309023022651672,
 'entropy_loss': -0.012702923268079758,
 'forward_dynamics_loss': 5.017355918884277,
 'inverse_dynamics_loss': 9.91663646697998,
 'mean_control_rewards': 0.00595580879598856,
 'mean_count_rewards': 0.9768397808074951,
 'mean_episode_return': 0.6924794316291809,
 'mean_intrinsic_rewards': 0.0029456191696226597,
 'mean_rewards': 0.02640078030526638,
 'mean_total_rewards': 0.029346399009227753,
 'pg_loss': 0.2990989089012146,
 'total_loss': 15.41347885131836}
```

However, in more harder environments,  I still cannot get a reward greater than 0.  
Is it because part of the code is missing (for example, observation normalization)?

**Comment by rraileanu** [comment]
One difference I can see in the arguments is the epsilon coefficient for RMSprop. The default is 1e-5 and it seems that you are using 0.01. Have you tried running the same experiment with args.epsilon = 1e-5?

**Comment by CrazySssst** [comment]
Thanks for your help, I got your results by setting the args.epsilon = 1e-5 in N7S4 and N12S10.
 I use 0.01 because  the epsilon coefficient for RMSprop is 0.01 in Table 2 in APPENDIX A.2 of RIDE [paper.](https://openreview.net/pdf?id=rkg-TJBFPB) 
Maybe I misunderstood Table 2 ? 
Anyway, thank you for your generous help

**Comment by rraileanu** [comment]
Glad to hear you are able to reproduce the results now!

Thank you for bringing this to my attention! I will update the paper with the correct value.
